source/develop/release-management/features/storage/4k-blocks.md

```diff
@@ -0,0 +1,165 @@
+---
+title: 4k drive support
+category: feature
+authors: dchaplyg
+feature_name: 4k drive support
+feature_modules: vdsm
+feature_status: Planning
+---
+
+# 4k drive support
+
+# Summary
+
+VDSM assumes, that 512 bytes is a minimally accessible block on the storage device, which can be written or read atomically. That was true for a long time, but now most of the storage devices changed their block size to the 4096 bytes (4 kilobytes). It is still compatible with VDSM, cause while we are requesting 512 atomic read or write, underlying storage will atomically read 4096 bytes, throw away 3584 bytes and return just 512. As it still acts same way, it has no impact on the VDSM. 
+
+Unfortunately, some 4k block storage devices either refuse to read/write smaller blocks atomically (some SAN appliances) or have a performance impact while emulating 512 bytes block device (like VDO). 
+
+Therefore without 4k block device support oVirt either can't use those devices as a storage domain or will have performance degradation on those domains.
+
+## Owner
+
+*   Name: [Denis Chaplygin](https://github.com/akashihi)
+*   Email: <dchaplyg@redhat.com>
+
+## Current status
+
+*   Target Release: 4.3
+*   Status: Prototyping
+*   Last updated: August 22, 2018
+
+# Implementation details
```

Comment:
I think we should avoid implementation details at this stage of development. What we need instead is description of what will be supported and what will not supported with 4k drives. Information that users will find useful.

For example, how a user will create a storage domain using 4k drives. How the system will be changed to support 4k drives should be at the end, under "implementation" section. At this point it is not important.